src/back_end.rs

```diff
@@ -10,79 +13,239 @@ const UV_COMPONENTS: usize = 2;
+        let stencil_inside = Stencil::new(Comparison::Equal, 255,
+            (StencilOp::Keep, StencilOp::Keep, StencilOp::Keep));
+        let stencil_outside = Stencil::new(Comparison::NotEqual, 255,
+            (StencilOp::Keep, StencilOp::Keep, StencilOp::Keep));
+
+        // Channel color masks.
+        let mask_all = gfx::state::MASK_ALL;
+        let mask_none = gfx::state::MASK_NONE;
+
+        let glsl = opengl.to_glsl();
+
+        let colored_shader_set = factory.create_shader_set(
+                Shaders::new()
+                    .set(GLSL::V1_20, colored::VERTEX_GLSL_120)
+                    .set(GLSL::V1_50, colored::VERTEX_GLSL_150_CORE)
+                    .get(glsl).unwrap(),
+                Shaders::new()
+                    .set(GLSL::V1_20, colored::FRAGMENT_GLSL_120)
+                    .set(GLSL::V1_50, colored::FRAGMENT_GLSL_150_CORE)
+                    .get(glsl).unwrap(),
+            ).unwrap();
+
+        let colored_pipeline = |factory: &mut F,
+                                blend_preset: Blend,
+                                stencil: Stencil,
+                                color_mask: gfx::state::ColorMask|
+        -> PipelineState<R, pipe_colored::Meta> {
+            factory.create_pipeline_state(
+                &colored_shader_set,
+                Primitive::TriangleList,
+                Rasterizer::new_fill(gfx::state::CullFace::Nothing),
+                pipe_colored::Init {
+                    pos: (),
+                    color: "color",
+                    blend_target: ("o_Color", color_mask, blend_preset),
+                    stencil_target: stencil,
+                    blend_ref: (),
+                    scissor: (),
+                }
             ).unwrap()
         };
 
-        let program_uv = {
-            let vertex = gfx::ShaderSource {
-                glsl_120: Some(textured::VERTEX_GLSL_120),
-                glsl_150: Some(textured::VERTEX_GLSL_150_CORE),
-                .. gfx::ShaderSource::empty()
-            };
-            let fragment = gfx::ShaderSource {
-                glsl_120: Some(textured::FRAGMENT_GLSL_120),
-                glsl_150: Some(textured::FRAGMENT_GLSL_150_CORE),
-                .. gfx::ShaderSource::empty()
-            };
-            factory.link_program_source(
-                vertex,
-                fragment
+        // Fake disabled blending using the same pipeline.
+        let no_blend = Blend {
+            color: BlendChannel {
+                equation: Equation::Add,
+                source: Factor::One,
+                destination: Factor::Zero,
+            },
+            alpha: BlendChannel {
+                equation: Equation::Add,
+                source: Factor::One,
+                destination: Factor::Zero,
+            },
+        };
+
+        let colored = PsoBlend {
+            alpha: colored_pipeline(factory, blend::ALPHA, stencil, mask_all),
```

Comment:
I guess you could generate the pipelines automatically for all the blend modes. I.e. pass in a function of `|blend| -> PipelineState` somewhere into PsoBlend constructor.